state/cluster/badger/mutator.go

```diff
@@ -36,202 +40,303 @@ func NewMutableState(state *State, tracer module.Tracer, headers storage.Headers
 	return mutableState, nil
 }
 
-// Extend validates that the given cluster block passes compliance rules, then inserts
-// it to the cluster state.
-// TODO (Ramtin) pass context here
-// Expected errors during normal operations:
-//   - state.OutdatedExtensionError if the candidate block is outdated (e.g. orphaned)
-//   - state.InvalidExtensionError if the candidate block is invalid
-func (m *MutableState) Extend(block *cluster.Block) error {
-
-	blockID := block.ID()
+// extendContext encapsulates all state information required in order to validate a candidate cluster block.
+type extendContext struct {
+	candidate                *cluster.Block // the proposed candidate cluster block
+	finalizedClusterBlock    flow.Header    // the latest finalized cluster block
```

Comment:
any strong reason for keeping `flow.Header` as value instead of pointer here?

Reply:
Changed to pointer in https://github.com/onflow/flow-go/pull/4148/commits/98dab1f5cee6b37e4d4c0698d03a519f51ea1c57